react/modules/push.ts

```diff
@@ -1,5 +1,23 @@
+import { createOrGetAnalyticsData } from './analytics'
+
 window.dataLayer = window.dataLayer || []
 
-export default function push(event: any) {
+export default function push(rawEvent: Record<string, unknown>) {
+  const {
+    location: originalLocation,
+    referrer: originalReferrer,
+    origin,
+  } = createOrGetAnalyticsData()
+
+  let event = rawEvent
+
+  if (window.dataLayer.length === 0 || origin === 'fresh') {
+    event = {
+      ...rawEvent,
+      originalLocation,
+      originalReferrer,
```

Comment:
This code changes the current event, overriding originalLocation and originalReferrer. This isn't a breaking change?

Reply:
It isn't. These properties are just ignored if users don't do anything with them. They essentially were already there, actually: the data between events are actually shared because the true thing is they're just appended over and over inside the same object (dataLayer object). So this shouldn't break anything.